pipeline/relation_embedder/path_concatenator/src/test/scala/weco/pipeline/path_concatenator/PathsServiceTest.scala

```diff
@@ -0,0 +1,193 @@
+        expectedWork,
+        work(path = "parent/child/grandchild")
+      )
+      withLocalMergedWorksIndex { index =>
+        insertIntoElasticsearch(index, works: _*)
+        whenReady(queryWorkWithPath(service(index), path = "parent/child")) {
+          _ shouldBe expectedWork
+        }
+      }
+    }
+
+    it("throws an exception if there are multiple matching exact paths") {
+      val works: List[Work[Merged]] = List(
+        work(path = "parent/child"),
+        work(path = "parent/child")
+      )
+      withLocalMergedWorksIndex { index =>
+        insertIntoElasticsearch(index, works: _*)
+
+        queryWorkWithPath(
+          service(index),
+          path = "parent/child"
+        ).failed.futureValue shouldBe a[RuntimeException]
+      }
+    }
+
+    it("throws an exception if no work can be found") {
+      val works: List[Work[Merged]] = List(
+        work(path = "hello/world")
+      )
+      withLocalMergedWorksIndex { index =>
+        insertIntoElasticsearch(index, works: _*)
+
+        queryWorkWithPath(
+          service(index),
+          path = "parent/child"
+        ).failed.futureValue shouldBe a[RuntimeException]
+      }
+    }
+  }
+
+  describe("The PathService childPath getter") {
+    it("Fetches works whose paths start with the end of this path") {
+      val works: List[Work[Merged]] = List(
+        work(path = "grandparent"),
+        work(path = "grandparent/parent"),
+        work(path = "parent/child/grandchild1"),
+        work(path = "parent/child/grandchild2"),
+        work(path = "grandparent/parent/child/grandchild3"), // ignored - it has already been resolved up to grandparent
+        work(path = "child/grandchild3") // ignored - could be child of a different parent
+      )
+      withLocalMergedWorksIndex { index =>
+        insertIntoElasticsearch(index, works: _*)
+        whenReady(queryChildWorks(service(index), path = "grandparent/parent")) {
+          _ should contain theSameElementsAs List(works(2), works(3))
+        }
+      }
+    }
+    it("Returns an empty list if there are no children") {
+      val works: List[Work[Merged]] = List(
+        work(path = "grandparent"),
+        work(path = "grandparent/parent")
+      )
+      withLocalMergedWorksIndex { index =>
+        insertIntoElasticsearch(index, works: _*)
+        whenReady(queryChildWorks(service(index), path = "grandparent/parent")) {
+          _ shouldBe empty
+        }
+      }
+    }
+  }
+
+  def queryParentPath(service: PathsService,
+                      childPath: String): Future[Option[String]] =
+    service.getParentPath(childPath)
+
+  def queryWorkWithPath(service: PathsService,
+                        path: String)(): Future[Work[Merged]] =
+    service.getWorkWithPath(path)
+
+  def queryChildWorks(service: PathsService,
+                      path: String): Future[Seq[Work[Merged]]] =
+    service.getChildWorks(path) // TODO: Add a test to show this works for more than 10 children
```

Comment:
Have you added this test, or is this a TODO for a separate PR?

Reply:
Sorry, yes, I had a couple of TODOs that I did, but forgot to actually remove the todo.